### ISSUE TITLE
torch.fft.fftfreq behaves unexpectedly when run on MPS backend

### ISSUE THREAD

**Issue by cooornm** [pytorch/pytorch#135223]
### 🐛 Describe the bug

When generating a frequency axis using `torch.fft.fftfreq(N)` on the MPS backend, the generated output is different from what the CPU produces. The result should be a tensor of length N, which starts at 0.0 and linearly increases to 0.5 at index N/2, then jumps to -0.5 and increases linearly again until -0.0001.

The below example demonstrates the different outputs when being run on CPU vs MPS.

```
import torch

device = torch.device('cpu')
freq_cpu = torch.fft.fftfreq(10**4, device=device).to('mps').cpu()
device = torch.device('mps')
freq_mps = torch.fft.fftfreq(10**4, device=device).cpu()

print('CPU: fftfreq ranges from %f to %f' % (freq_cpu.min(), freq_cpu.max()) )
print('MPS: fftfreq ranges from %f to %f' % (freq_mps.min(), freq_mps.max()) )
```

### Versions

PyTorch version: 2.4.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.2.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.1.0.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.6 (default, Nov 10 2023, 13:38:27)  [Clang 15.0.0 (clang-1500.1.0.2.5)] (64-bit runtime)
Python platform: macOS-14.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1

Versions of relevant libraries:
[pip3] numpy==2.0.1
[pip3] torch==2.4.0
[pip3] torchaudio==2.4.0
[conda] Could not collect

cc @mruberry @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by malfet** [comment]
Could not reproduce it using the local build running on M2 Pro:
```
% python -c "import torch;print(torch.__version__, (torch.fft.fftfreq(10**4, device='mps').cpu()-torch.fft.fftfreq(10**4)).abs().max())"
2.5.0a0+git9158bb7 tensor(0.)
```

**Comment by hvaara** [comment]
@cooornm are you still able to reproduce if you use latest nightly?

I was able to repro bad at `v2.4.1` and good at `main` (a6fae2e).

Running a bisect to find the hero commit.

**Comment by hvaara** [comment]
Hero commit identified as 861bdf96f4dc7cbdeadd65617ed5059e4030113d (PR #128393)

#### Bisect replay log (note that good and bad are inverted)
```bash
git bisect start
# status: waiting for both good and bad commits
# good: [ee1b6804381c57161c477caa380a840a84167676] [Doc] Fix rendering of the unicode characters (#134695)
git bisect good ee1b6804381c57161c477caa380a840a84167676
# status: waiting for bad commit, 1 good commit known
# bad: [042f2f7746a064f1527d95d1f1d712b4f0b34186] [ONNX] Re-raise the exception if the dynamic shapes cannot be refined (#135418)
git bisect bad 042f2f7746a064f1527d95d1f1d712b4f0b34186
# good: [b66e3f0957b96b058c9b632ca60833d9717a9d8a] Set simdlen based on ATEN_CPU_CAPABILITY (#123514)
git bisect good b66e3f0957b96b058c9b632ca60833d9717a9d8a
# good: [2988d33c8052fb065dca9f812f56c95795d46609]  [3/N] Fix clang-tidy warnings in jit  (#131830)
git bisect good 2988d33c8052fb065dca9f812f56c95795d46609
# good: [2cffe82deae5a0a075f817c4c706906f227e3244] Fix triton build failure due to tritonlang.blob.core.windows.net not available (#133694)
git bisect good 2cffe82deae5a0a075f817c4c706906f227e3244
# bad: [9d0e0e6f1def53796f4adab9a9e3b3b0e220f388] [inductor] calibration inductor windows uts (14/N) (#134585)
git bisect bad 9d0e0e6f1def53796f4adab9a9e3b3b0e220f388
# bad: [64cfcbd8a318535560b8fc908a0948af8e177329] Tune _int_bsr_dense_addmm for int8 inputs on A100 (#134035)
git bisect bad 64cfcbd8a318535560b8fc908a0948af8e177329
# bad: [7492da804fbcaf5fe983d4be154aa0263b68e693] Mark disabled tests as fixed (#133940)
git bisect bad 7492da804fbcaf5fe983d4be154aa0263b68e693
# bad: [c0c82a5f6a06aecb8310400ae0f407a8f495b573] [CUDA][SDPA] Bump tolerances for `test_mem_efficient_attention_attn_mask_vs` (#133738)
git bisect bad c0c82a5f6a06aecb8310400ae0f407a8f495b573
# bad: [b0803129e8cb5813b927f6b3fea846bf2dd58a6f] Added meta registration for `_fused_adamw_` (#133728)
git bisect bad b0803129e8cb5813b927f6b3fea846bf2dd58a6f
# good: [41e6619509401c120b694342352ae84b94df6511]  [codemod] Del un at::native::metal @ MPSCNNFullyConnectedOp.h:6 (export D59157302) (#133515)
git bisect good 41e6619509401c120b694342352ae84b94df6511
# bad: [fd33499b0c355c42a62cd9628c89b543fb142926] [PT2][Optimus] Fix mixed precison training problem in decompose mem bound (#133626)
git bisect bad fd33499b0c355c42a62cd9628c89b543fb142926
# good: [f57b00704e498a676854a02974ca9e0c42188b23] [Traceable FSDP2][Dynamo] Support reconstructing CUDA event object within Dynamo graph (#133635)
git bisect good f57b00704e498a676854a02974ca9e0c42188b23
# bad: [861bdf96f4dc7cbdeadd65617ed5059e4030113d] [MPS] Add native strided API for MPSNDArray starting with macOS 15 (#128393)
git bisect bad 861bdf96f4dc7cbdeadd65617ed5059e4030113d
# good: [447f428d6d6563636f95edae3fd24b42038c9ba2] [ROCm] Fix text_export cudnn_attention UT (#133234)
git bisect good 447f428d6d6563636f95edae3fd24b42038c9ba2
# first bad commit: [861bdf96f4dc7cbdeadd65617ed5059e4030113d] [MPS] Add native strided API for MPSNDArray starting with macOS 15 (#128393)
```

**Comment by hvaara** [comment]
The newly added regression test from #135440 closed this. Please feel free to reopen if you're still able to reproduce or there are items left to address.

**Comment by atalman** [comment]
Validated. This is resolved in 2.5.1 final rc:
```
Output of the test:
CPU: fftfreq ranges from -0.500000 to 0.499900
MPS: fftfreq ranges from -0.500000 to 0.499900
```
